src/awkward/_util.py

```diff
@@ -418,6 +418,13 @@ def wrap(content, behavior):
         return content
 
 
+def maybe_wrap(content, behavior, highlevel):
+    if highlevel:
+        return ak._util.wrap(content, behavior)
+    else:
+        return content
```

Comment:
This could be used in a _lot_ of different places. Once this PR is merged, I can do a sweep to apply it consistently.

I think this kind of thing is on the borderline between DRY and WET, since it's such a small number of lines that it could be considered an "idiom," but the word "maybe_" in the function name makes it super-clear what this does (the tradeoff between indirection and repetition is not too bad on the indirection side: a reader of the code probably wouldn't have to look it up to see what "maybe_wrap" does).